R/sf.R

```diff
@@ -48,14 +50,23 @@ as_points <- function(df, xvar = "x", yvar = "y", remove = FALSE) {
     assert_that(has_name(df, xvar))
     assert_that(has_name(df, yvar))
 
+    x_var <- parse_expr( xvar )
+    y_var <- parse_expr( yvar )
     df_cleaned <-
-        df[!is.na(df[,xvar]) & !is.na(df[,yvar]),]
+        df %>%
+        filter(!is.na(!!xvar) & !is.na(!!yvar))
 
     if (nrow(df_cleaned) < nrow(df)) {
         warning(nrow(df) - nrow(df_cleaned),
                 " locations were removed because of missing X or Y coordinates.")
     }
 
+    check_dubbels <- df_cleaned %>% distinct(!!xvar, !!yvar)
+    if (nrow(check_dubbels) < nrow(df)) {
+        warning(nrow(df) - nrow(check_dubbels),
+                " locations are duplicated.")
+    }
```

Comment:
Thanks for the idea. A few thoughts however:

- about this warning: is it sensible to call duplicated coordinates 'duplicated locations'? Different locations in _Watina_ refer to different values of `loc_code` (i.e. effectively different devices); XY is just a spatial attribute. Also in other functions, `loc_code` (and associated `loc_wid`) plays the pivot role. It seems quite common for (close) locations that they can share the same XY-coordinates (see counts below). I'm rather inclined not to warn for duplicated "locations" in that case.

    If you still want the user to be notified, you could better say: 'Note that duplicated coordinates are present'. But I'm not sure if it is so needed - it is clear from the resulting data.

```r
library(tidyverse)
library(watina) # using inbo/watina@rc0.3
> get_locs(watina, 
+          filterdepth_range = c(0, 500), 
+          filterdepth_na = TRUE) %>% 
+     count
# Source:   lazy query [?? x 1]
# Database: Microsoft SQL Server
#   13.00.5598[INBO\floris_vanderhaeghe@INBO-SQL08-PRD\DWH/W0002_00_Watina]
      n
  <int>
1  9287
> get_locs(watina, 
+          filterdepth_range = c(0, 500), 
+          filterdepth_na = TRUE) %>% 
+     distinct(loc_code) %>% 
+     count
# Source:   lazy query [?? x 1]
# Database: Microsoft SQL Server
#   13.00.5598[INBO\floris_vanderhaeghe@INBO-SQL08-PRD\DWH/W0002_00_Watina]
      n
  <int>
1  9287
> get_locs(watina, 
+          filterdepth_range = c(0, 500), 
+          filterdepth_na = TRUE) %>% 
+     distinct(x, y) %>% 
+     count
# Source:   lazy query [?? x 1]
# Database: Microsoft SQL Server
#   13.00.5598[INBO\floris_vanderhaeghe@INBO-SQL08-PRD\DWH/W0002_00_Watina]
      n
  <int>
1  8216
```

- use English object names, eg `check_duplicates` instead of `check_dubbels`
- why first use `df_cleaned` and later (twice) use `nrow(df)`? I suggest to use `nrow(df_cleaned)`

Reply:
you've right it's not essential information, but sometimes it can be useful when one have to be sure there aren't any duplicates. 
I also agree that sometimes multiple locations can have the same coordinates (eg a set of tubes with different lengths). So it's better to speak of rows and use English object names (;-)